packages/expo-cli/src/commands/build-native/prepare.ts

```diff
@@ -2,37 +2,50 @@ import { Android, BuildConfig, BuildType, Job, JobSchema, Platform, iOS } from '
 
 export type Options = {
   platform: Platform;
-  type: BuildType;
 };
 
-async function prepareJob(options: Options, projectUrl: string, projectDir: string): Promise<Job> {
-  const credentials = await getPlatformCredentials(options.platform, projectDir);
-  const rawJob = {
-    credentials,
-    platform: options.platform,
-    projectUrl,
-    type: options.type,
-  };
-
-  const { value, error } = JobSchema.validate(rawJob);
+function validateJob(job: Job): Job {
+  console.log(job);
+  const { value, error } = JobSchema.validate(job);
   if (error) {
     throw error;
   } else {
-    return value as Job;
+    return value;
   }
 }
 
-async function getPlatformCredentials(
-  platform: Platform,
-  projectDir: string
-): Promise<Android.Credentials | iOS.Credentials> {
-  const jsonCredentials = await BuildConfig.read(projectDir);
-  if (platform === Platform.Android) {
-    return await BuildConfig.prepareAndroidJobCredentials(jsonCredentials);
-  } else if (platform === Platform.iOS) {
-    return await BuildConfig.prepareiOSJobCredentials(jsonCredentials);
-  } else {
-    throw new Error('Unsupported platform');
+async function prepareJob(options: Options, projectUrl: string, projectDir: string): Promise<Job> {
+  const turtleJson = await BuildConfig.read(projectDir);
+
+  switch (options.platform) {
```

Comment:
Can we move this switch to `@expo/build-tools`? I don't see a reason to keep it here.
If it was in `@expo/build-tools` we could only export one function from there.

Reply:
i think it that it can make a difference later, manged build will require manifest
Do we want tuertle to be aware of "shape" of the project?